### DISCUSSION TITLE
What happens to my run / partition history if I switch the partitions definition associated with my job / asset?

### DISCUSSION THREAD

**Discussion by dpeng817** [dagster-io/dagster#13991] [Q&A] [answered]
A la I switch from a static partitions definition to a dynamic partitions definition.

**Comment by sryza** [comment, marked as answer]
Only partition keys that are part of an asset's current `PartitionsDefinition` will be shown in:
- The Partitions tab of the Asset Details page
- The # missing / # materialized counts on the asset nodes in the asset graph

This means that any partition keys that no longer exist in the new `PartitionsDefinition` will no longer be represented in these places. If you change your `PartitionsDefinition` from a `StaticPartitionsDefinition` to a `DynamicPartitionsDefinition` or vice versa, and the keys remain the same, then it should not affect what you see in these places.

Runs that correspond to partitions that are not part of the current `PartitionsDefinition` will still show up on the Runs page and Events tab of the Asset Details page.
